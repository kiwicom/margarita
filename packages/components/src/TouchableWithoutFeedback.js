// @flow strict

import * as React from 'react';
import { TouchableWithoutFeedback as OriginalTouchableWithoutFeedback } from 'react-native'; // eslint-disable-line no-restricted-imports
import type { StylePropType } from '@kiwicom/universal-components';

type Props = {|
  +onPress: () => void,
  +children: React.Node,
  +accessibilityTraits: string,
  +accessibilityComponentType: string,
  +style?: StylePropType,
  +onMouseEnter?: () => void,
  +onMouseLeave?: () => void,
|};

export default function TouchableWithoutFeedback(props: Props) {
  return <OriginalTouchableWithoutFeedback {...props} />;
}

TouchableWithoutFeedback.defaultProps = {
  accessibilityTraits: 'button',
  accessibilityComponentType: 'button',
};
