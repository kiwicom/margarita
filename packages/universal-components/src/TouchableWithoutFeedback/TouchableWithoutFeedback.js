// @flow strict

import * as React from 'react';
import { TouchableWithoutFeedback as OriginalTouchableWithoutFeedback } from 'react-native';
import type { StylePropType } from '@kiwicom/universal-components';

type Props = {|
  +onPress: () => void,
  +disabled?: boolean,
  +children: React.Node,
  +accessibilityTraits: string,
  +accessibilityComponentType: string,
  +style?: StylePropType,
  +hitSlop?: {
    left: number,
    right: number,
    top: number,
    bottom: number,
  },
  +onMouseEnter?: () => void,
  +onMouseLeave?: () => void,
|};

export default function TouchableWithoutFeedback(props: Props) {
  return <OriginalTouchableWithoutFeedback {...props} />;
}

TouchableWithoutFeedback.defaultProps = {
  accessibilityTraits: 'button',
  accessibilityComponentType: 'button',
};
