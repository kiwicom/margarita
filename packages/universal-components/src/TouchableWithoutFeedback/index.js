// @flow

export {
  default as TouchableWithoutFeedback,
} from './TouchableWithoutFeedback';
