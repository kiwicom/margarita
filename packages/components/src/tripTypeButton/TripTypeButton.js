// @flow

import * as React from 'react';
import { View } from 'react-native';
import { StyleSheet, Text, Icon } from '@kiwicom/universal-components';
import { defaultTokens } from '@kiwicom/orbit-design-tokens';

import TouchableWithoutFeedback from '../TouchableWithoutFeedback';

type Props = {|
  +icon: string,
  +label: string,
  +onPress: () => void,
|};

export default class TripTypeButton extends React.Component<Props> {
  render() {
    const { onPress, icon, label } = this.props;
    return (
      <TouchableWithoutFeedback onPress={onPress}>
        <View style={styles.container}>
          <Icon color={defaultTokens.colorIconSecondary} name={icon} />
          <Text style={styles.label}>{label}</Text>
          <Icon color={defaultTokens.colorIconSecondary} name="show-more" />
        </View>
      </TouchableWithoutFeedback>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    height: parseInt(defaultTokens.heightButtonNormal, 10),
    backgroundColor: defaultTokens.backgroundButtonSecondary,
    borderRadius: parseInt(defaultTokens.borderRadiusLarge, 10),
    paddingHorizontal: 11,
  },
  label: {
    flex: 1,
    color: defaultTokens.paletteInkDark,
    fontSize: parseInt(defaultTokens.fontSizeButtonNormal, 10),
    marginLeft: 10,
  },
});
