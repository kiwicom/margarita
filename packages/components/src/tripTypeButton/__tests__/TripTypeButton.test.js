// @flow

import * as React from 'react';
import { shallow, render, fireEvent } from 'react-native-testing-library';

import TripTypeButton from '../TripTypeButton';
import TouchableWithoutFeedback from '../../TouchableWithoutFeedback';

const icon = 'flight-direct';
const label = 'One way';
const onPress = jest.fn();

const { getByType } = render(
  <TripTypeButton icon={icon} label={label} onPress={onPress} />,
);

it('should execute onPress method', () => {
  fireEvent(getByType(TouchableWithoutFeedback), 'press');
  expect(onPress).toHaveBeenCalledWith(undefined);
});

it('renders', () => {
  expect(
    shallow(<TripTypeButton icon={icon} label={label} onPress={onPress} />),
  ).toMatchSnapshot();
});
