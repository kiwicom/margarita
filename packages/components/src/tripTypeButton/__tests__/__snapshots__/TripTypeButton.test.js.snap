// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Object {
  "output": <TouchableWithoutFeedback
    accessibilityComponentType="button"
    accessibilityTraits="button"
    onPress={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#e8edf1",
          "borderRadius": 6,
          "flex": 1,
          "flexDirection": "row",
          "height": 44,
          "paddingHorizontal": 11,
        }
      }
    >
      <Icon
        color="#7f91a8"
        name="flight-direct"
      />
      <Text
        style={
          Object {
            "color": "#171b1e",
            "flex": 1,
            "fontSize": 14,
            "marginLeft": 10,
          }
        }
      >
        One way
      </Text>
      <Icon
        color="#7f91a8"
        name="show-more"
      />
    </Component>
  </TouchableWithoutFeedback>,
}
`;
